src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -410,7 +409,15 @@ private boolean updateMutedComponents() {
 
       LOG.info("Updating the muted graph nodes to : {}", graphNodesForMute);
       Stats.getInstance().updateMutedGraphNodes(graphNodesForMute);
+      // We need to update muted actions in two places. One is in the controller which could
+      // potentially serve as a snapshot when creating a new rca scheduler, and the other place
+      // is in the rca scheduler itself. The scheduler maintains a snapshot of the appcontext(see
+      // the comment above near scheduler instantiation) that it uses to determine staleness. It
+      // also happens to be the appcontext that will be passed to the graph nodes.
       appContext.updateMutedActions(actionsForMute);
+      if (rcaScheduler != null) {
+        rcaScheduler.updateAppContextWithMutedActions(actionsForMute);
+      }
```

Comment:
This fixes this issue but we can run into such inconsistencies where changes may not reflect until a scheduler restart. Can the scheduler use a reference of the AppContext of RcaController instead of a copy ?

Reply:
That was my first thought as well. But I saw this comment on L215 and decided to go this way.
> // RcaScheduler should be started with a snapshot of the AppContext as RcaController
      // monitors it for stale state and always restarts the scheduler if it finds its state
      // stale.

I think you have the best context regarding AppContext usage, if this comment is stale, then I can go ahead and use the same AppContext reference in both the controller and the scheduler.

Reply:
We will have to handle the case of changed role for the RCAScheduler. we can do the cleanup for a follow up PR. This is a bug fix, so I won't block this on the proper fix.